cli/src/commands/start.rs

```diff
@@ -521,15 +521,6 @@ impl Start {
         // Print the welcome.
         println!("{}", crate::helpers::welcome_message());
 
-        // Check if we are running with the lower coinbase and proof targets. This should only be
-        // allowed in --dev mode and should not be allowed in mainnet mode.
-        if cfg!(feature = "test_network") && self.dev.is_none() {
```

Comment:
We can and should keep the check in place that we can only use this in dev mode

Reply:
The original code is intact on the latest commit, we are bailing and exiting and I assume the error is reported in CLI

Reply:
I mean instead of removing the if branch entirely, change it to use `warn!` instead of `bail!`

Reply:
More general, I like the idea of surfacing the feature flags and network. We already collect and print this for the --version flag, not sure if its a log line. Feel free to add in a separate PR.